pkg/controllermanager/apis/config/types.go

```diff
@@ -127,6 +130,16 @@ type ProjectControllerConfiguration struct {
 	StaleSyncPeriod *metav1.Duration
 }
 
+// QuotaConfiguration defines quota configurations.
+type QuotaConfiguration struct {
+	// ProjectSelector is an optional setting to select the projects considered for quotas.
+	// Defaults to empty LabelSelector, which matches everything.
+	ProjectSelector *metav1.LabelSelector
+	// ResourceQuota is the quota specification used for the project set-up.
+	// Only v1.ResourceQuota resources are supported.
+	ResourceQuota runtime.Object
```

Comment:
Why is this a `runtime.Object`? Is it not possible to use `v1.ResourceQuota` directly?

Reply:
It's just used as proposed in the `runtime.RawExtension` [doc string](https://github.com/kubernetes/apimachinery/blob/3a3fc754224fdaf5be79aa6583c78d1f5e411930/pkg/runtime/types.go#L49).

Reply:
But `ResourceQuota` is not really an "extension object", we already know which object is supposed to be in that field, so do we actually need `runtime.RawExtension` here?

Reply:
Maybe not an extension object but a place holder for any external version which makes the API independent from the implementation. Hence, we are open to support further quota GVKs in the future. I don't see a disadvantage as long as it is documented.

Reply:
Ah yep. That makes sense. Thanks for explaining!